src/Voron/Platform/Win32/Win32MemoryMapPager.cs

```diff
@@ -436,5 +454,49 @@ public override void MaybePrefetchMemory(List<TreePage> sortedPages)
                     entries, 0);
             }
         }
+
+        internal override void ProtectPageRange(byte* start, ulong size, bool force = false)
+        {
+            if (UsePageProtection || force)
+            {
+                Win32NativeMethods.MEMORY_BASIC_INFORMATION memoryInfo1 = new Win32NativeMethods.MEMORY_BASIC_INFORMATION();
+                int vQueryFirstOutput = Win32NativeMethods.VirtualQuery(start, &memoryInfo1, new UIntPtr(size));
+                int vQueryFirstError = Marshal.GetLastWin32Error();
+
+                Win32NativeMethods.MemoryProtection oldProtection;
+                bool status = Win32NativeMethods.VirtualProtect(start, new UIntPtr(size), Win32NativeMethods.MemoryProtection.READONLY, out oldProtection);
+                if (!status)
+                {
+                    int vProtectError = Marshal.GetLastWin32Error();
+
+                    Win32NativeMethods.MEMORY_BASIC_INFORMATION memoryInfo2 = new Win32NativeMethods.MEMORY_BASIC_INFORMATION();
+                    int vQuerySecondOutput = Win32NativeMethods.VirtualQuery(start, &memoryInfo2, new UIntPtr(size));
+                    int vQuerySecondError = Marshal.GetLastWin32Error();
+                    Debugger.Break();
```

Comment:
Debugger.Break while an interesting thing to happen, will have impact on CI server if we have such an error.